### PR TITLE
Change to new topics and sensors

### PR DESCRIPTION
**Pull request by Blake-Haydon**
## Description

**Most important changes**
+ Converted topics into an enum file for easier readability and decreased hardcoding
+ Fix changed topics eg. /v3/wireless-module/<id>/low-battery. Now all of the code matches the spec on notion
+ Added new sensors (power, cadence and heartRate)
+ Added support for multiple mock sensors (any number can now be specified eg. 20)
+ Removed huge if statements to define the modules, now it is simply a function and you can specify what sensors you want in a list

**Less important changes** 
+ Fixed some documentation so now it makes sense
+ If statements now check the entire topic rather than just the start and the end
+ `if None` hack to specify all sensors has been removed so now all the sensors that are being mocked are just stored in an array
+ Saved file names are now generated by the logging script in the format `module_id + "_D%d-%m-%d_T%H-%M-%S" + ".csv"`

## Screenshots

image
As many sensors as you need / want

## Steps to Test
1. Make sure mosquitto is running
2. Start logger `python3 mqtt_wireless_log.py`
3. Start mock modules `python3 mock_wireless_module.py -t 5 --id 1 2 3 4 5 666`

Watch as 6 fake modules generate and save data for 5 seconds

### CODE REVIEW

File: Raspi/Test/DataToTempCSV.py [hunk 1]
```diff
@@ -66,23 +67,26 @@ def make_temp_csv():
     module_data = json.loads(module_data)
 
     # Determine which type of data to parse
-    if msg.topic.endswith("data"):
+    if enum_topics.WirelessModule.data(module_id_num) == msg.topic:
         type = "DATA"
```
Comment (emilytrau) [inline Raspi/Test/DataToTempCSV.py]: Message type could also be an enum

File: Raspi/Test/DataToTempCSV.py [hunk 2]
```diff
@@ -42,7 +43,7 @@ def make_temp_csv():
 
         # Ensures that the temp file is in the same folder as the script
         current_dir = os.path.dirname(__file__)
-        temp_filename = '.~temp_' + module_id + '_' + type + '.csv'
+        temp_filename = '.~temp_' + module_id_str + '_' + type + '.csv'
```
Comment (emilytrau) [inline Raspi/Test/DataToTempCSV.py:46]: This could be an fstring

File: Raspi/Test/mock_wireless_module.py [hunk 3]
```diff
@@ -64,98 +109,55 @@ def send_fake_data(client, duration, rate, module_id):
...
-                                    "type": "gps",
-                                    "value": s_gps.get_value()
-                                }
-                             ]
-                          }
-            publish_data_and_battery(module_num)
+        print("TIME:", current_time)
+        for module_id_num in module_id_nums:
+            # Wireless module 1 (Front)
+            if module_id_num == 1:
```
Comment (emilytrau) [inline Raspi/Test/mock_wireless_module.py:132]: If you declared the modules in a list or dictionary you wouldn't need this lookup anymore

**Review by emilytrau — CHANGES_REQUESTED** [review]

**Review by khanguslee — COMMENTED** [review]
What an amazing Pull Request description. Really appreciate it. Pretty much got nothing else except for the already mentioned changes.

File: Raspi/Test/enum_topics.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+
+class WirelessModule(NeatEnum):
+    """Building blocks to make a topic for a wireless module. All that is
+    needed is to add a <module_id> number."""
+
+    base = '/v3/wireless-module/'
+
+    def battery(module_id):
+        return '/v3/wireless-module/' + str(module_id) + '/battery'
```
Comment (hallgchris) [inline Raspi/Test/enum_topics.py]: Did you mean to reference `base` here rather than hardcode the string? Could also use an fstring.

**Review by hallgchris — COMMENTED** [review]
Just one thing! Looks great

**Review by emilytrau — DISMISSED** [review]

**Review by khanguslee — APPROVED** [review]
LGTM. The filenames should not be generated by time as there is not RTC on the Bike DAS (for now). Can you generate filenames as per the old logger. Happy to have that as a new PR. But please make it a priority for now.

**Review by hallgchris — APPROVED** [review]

**Review by emilytrau — APPROVED** [review]

**Pull request merged**